test/protocol.js

```diff
@@ -78,6 +72,12 @@ test('Asynchronous handshake + extended handshake', t => {
     if (ext === 'handshake') {
       eventLog.push('w2 ex')
       t.ok(obj)
+
```

Comment:
these tests operated on process.nextTick being fucky, and checked if w2ex was LAST, when pushing a 2nd to last entry, and hoped "yeah lets hope there's no async here, otherwise this falls apart" jesus, why???